azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/Vault.java

```diff
@@ -204,6 +236,13 @@
              * @return the next stage of key vault definition
              */
             WithCreate withTemplateDeploymentDisabled();
+            
+            /**
+             * Set the createMode value. 
+             * 
+             * @return the next stage of key vault definition
+             */
+            WithCreate withCreateMode(CreateMode createMode);
```

Comment:
@tiffanyachen Could CreateMode.RECOVER be used with an existing Key Vault resource that has not been deleted? What impact other settings prior to reaching this method during the create flow will have when this value is set?

Reply:
@tiffanyachen If this is the case then we will have to rework a bit the "Create()" flow. What's the maximum properties in the JSON body that can be passed to the create op when CreateMode.RECOVER is specified?
Basically we will need to think of a way to shortcut the current create flow for the user to be able to use this setting. I'm thinking it might be worth having a separate set of methods as part of Vaults.java something like recoverDeletedVault(resourceGroupName, vaultName) or recoverSoftDeletedVault() (plus the Async one), instead of expecting the user to specify the CreateMode via the define()/create() flow... Because I don't see the later how it will ever succeed when we ask the user for SKU and other stuff before it can reach to withCreateMode().